decidim-core/lib/decidim/content_security_policy.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This class is responsible of generating the Content-Security-Policy header
+  # for the application. It takes into account the organization's CSP settings
+  # and the additional settings defined in the initializer.
+  class ContentSecurityPolicy
+    SUPPORTED_POLICIES = %w(
+      child-src
+      connect-src
+      default-src
+      font-src
+      frame-src
+      img-src
+      manifest-src
+      media-src
+      object-src
+      prefetch-src
+      script-src
+      script-src-elem
+      script-src-attr
+      style-src-elem
+      style-src-attr
+      worker-src
+      base-uri
+      sandbox
+      form-action
+      frame-ancestors
+      navigate-to
+      report-uri
+      report-to
+      require-trusted-types-for
+      trusted-types
+      upgrade-insecure-requests
+      style-src
+    ).freeze
+
+    def initialize(organization = nil, additional_policies = {})
+      @organization = organization
+      @policy = default_policy
+      @additional_policies = additional_policies
+    end
+
+    def output_policy
+      add_system_csp_directives
+      add_additional_policies
+      organization_csp_directives
+
+      format_policies
+    end
+
+    def append_csp_directive(directive, value)
+      return if value.blank?
+
+      unless SUPPORTED_POLICIES.include?(directive)
+        Rails.logger.debug { "Invalid Content Security Policy directive: #{directive}, supported directives: #{SUPPORTED_POLICIES.join(", ")}" }
```

Comment:
Just to open a discussion with other @decidim/maintainers, but I'd expect for this error to be a  raise instead of a log ("fail fast" aka "failing immediately and visibly"), at least in the case that this change is made by an implementer/developer (i.e. by changing the initializer code). 
If the case is that its changed by an implementer through the System panel, then I'd expect to have these kind of errors as validations in the form. 

(I still need to give it a spin in my machine and actually try this code out, so you can ignore this comment for now until after I actually finish trying this out 😅)